evennia/utils/tests/test_validatorfuncs.py

```diff
@@ -0,0 +1,139 @@
+"""Tests for validatorfuncs """
+
+from django.test import TestCase
+from evennia.utils import validatorfuncs
+import mock
+import datetime
+import pytz
+
+
+class TestValidatorFuncs(TestCase):
+
+    def test_text_ok(self):
+        for val in [None, -123, 'abc', 1.234, {1:True, 2:False}, ['a', 1]]:
+            self.assertEqual(str(val), validatorfuncs.text(val))    
+
+    @mock.patch('builtins.str')
+    def test_text_raises_ValueError(self, mocked_str):
+        mocked_str.side_effect = Exception
+        with self.assertRaises(ValueError):
+            validatorfuncs.text(None)
+
+    def test_color_ok(self):
+        for color in ['r', 'g', 'b', 'H', 'R', 'M', '^']:
+          self.assertEqual(color, validatorfuncs.color(color))
+
+    def test_color_falsy_raises_ValueError(self):
+        for color in [None, (), [], False, True, {}]:
+            with self.assertRaises(ValueError):
+                validatorfuncs.color(color)
+
+    def test_datetime_ok(self):
+        for dt in ['Oct 12 1:00 1492', 'Jan 2 12:00 2020', 'Dec 31 00:00 2018']:
+            self.assertTrue(
+                isinstance(validatorfuncs.datetime(dt, from_tz=pytz.UTC),
+                datetime.datetime))
+
+    def test_datetime_raises_ValueError(self):
+        for dt in ['', 'January 1, 2019', '1/1/2019', 'Jan 1 2019']:
+            with self.assertRaises(ValueError):
+                validatorfuncs.datetime(dt)
+
+    def test_duration_ok(self):
+        for d in ['1d', '2w', '3h', '4s', '5m', '6y']:
+            self.assertTrue(
+                isinstance(validatorfuncs.duration(d), datetime.timedelta))
+
+        # THE FOLLOWING FAILS, year calculation seems to be incorrect
+        # self.assertEqual(
+        #     datetime.timedelta(1+5*365, 2, 0, 0, 3, 4, 5),
+        #     validatorfuncs.duration('1d 2s 3m 4h 5w 5y'))
+
+    def test_duration_raises_ValueError(self):
+        for d in ['', '1', '5days', '1Week']:
+            with self.assertRaises(ValueError):
+                validatorfuncs.duration(d)
+
+    # def test_future_ok(self):
```

Comment:
There seems to be some issues with the creation of datetimes and their respective timezone. Some are coming up as naive.

I'll find the root cause and submit another PR with the fixes and these 2 tests enabled.

Reply:
Root cause identified and addressed. All tests pass.